ArcGISRuntimeSDKQt_CppSamples/Routing/NavigateARouteWithRerouting/NavigateARouteWithRerouting.cpp

```diff
@@ -0,0 +1,367 @@
+  m_mapView->locationDisplay()->setDataSource(m_simulatedLocationDataSource);
+  m_simulatedLocationDataSource->start();
+}
+
+void NavigateARouteWithRerouting::connectRouteTrackerSignals()
+{
+  connect(m_routeTracker, &RouteTracker::newVoiceGuidance, this, [this](VoiceGuidance* rawVoiceGuidance)
+  {
+    auto voiceGuidance = std::unique_ptr<VoiceGuidance>(rawVoiceGuidance);
+    m_speaker->say(voiceGuidance->text());
+  });
+
+  connect(m_routeTask, &RouteTask::solveRouteCompleted, this, [this](QUuid, RouteResult routeResult)
+    {
+      if (routeResult.isEmpty())
+        return;
+
+      if (routeResult.routes().empty())
+        return;
+
+      m_routeResult = routeResult;
+      m_route = qAsConst(m_routeResult).routes()[0];
+      m_directionManeuvers = m_route.directionManeuvers(this)->directionManeuvers();
+  });
+
+  connect(m_routeTracker, &RouteTracker::trackingStatusChanged, this, [this](TrackingStatus* rawTrackingStatus)
+  {
+    auto trackingStatus = std::unique_ptr<TrackingStatus>(rawTrackingStatus);
+    QString textString("Route status: \n");
+    if (trackingStatus->destinationStatus() == DestinationStatus::NotReached || trackingStatus->destinationStatus() == DestinationStatus::Approaching)
+    {
+      if (trackingStatus->isOnRoute())
+      {
+        textString += "Distance remaining: " + trackingStatus->routeProgress()->remainingDistance()->displayText() + " " +
+            trackingStatus->routeProgress()->remainingDistance()->displayTextUnits().pluralDisplayName() + "\n";
+        QTime time = QTime::fromMSecsSinceStartOfDay(trackingStatus->routeProgress()->remainingTime() * 60 * 1000); // convert time to milliseconds
+        textString += "Time remaining: " + time.toString("hh:mm:ss") + "\n";
+
+        // display next direction
+        if (m_directionManeuvers.size() > 0)
+        {
+          if (trackingStatus->currentManeuverIndex() + 1 < m_directionManeuvers.length())
+          {
+            textString += "Next direction: " + m_directionManeuvers[trackingStatus->currentManeuverIndex() + 1].directionText() + "\n";
+          }
+        }
+        m_routeTraveledGraphic->setGeometry(trackingStatus->routeProgress()->traversedGeometry());
+        m_routeAheadGraphic->setGeometry(trackingStatus->routeProgress()->remainingGeometry());
+      }
+    }
+    else if (trackingStatus->destinationStatus() == DestinationStatus::Reached)
+    {
+      textString += "Destination reached.\n";
+
+      // set the route geometries to reflect the completed route
+      m_routeAheadGraphic->setGeometry(Geometry());
+      m_routeTraveledGraphic->setGeometry(trackingStatus->routeResult().routes()[0].routeGeometry());
+
+      // navigate to next stop, if available
+      if (trackingStatus->remainingDestinationCount() > 1)
+      {
+        m_routeTracker->switchToNextDestination();
+      }
+      else
+      {
+        m_simulatedLocationDataSource->stop();
```

Comment:
Reenable the navigation button when the destination is reached
```suggestion
        m_simulatedLocationDataSource->stop();
        m_navigationEnabled = true;
        emit navigationEnabledChanged();
```